src/builtin/tables.scrbl

```diff
@@ -217,152 +345,294 @@ check:
-@red-method["reduce"
-  #:contract (a-arrow (apply Red-of Red-params) "Acc" "InVal" (a-tuple "Acc" "OutVal"))
-  #:args '(("self" #f) ("accumulator" #f) ("value-from-column" #f))
-  #:return (a-tuple "Acc" "OutVal")]
 
+@subsection{Reducers}
 
-Reducers are essentially descriptions of folds (in the list @pyret{fold}
-sense) over table columns. The way reducers are called by the language
-runtime is as follows: the value(s) from the first row are passed to the
-reducer's @pyret{.one} method, which should return a tuple containing both
-any accumulated information needed for the fold and the value which should
-be placed in the new column in that row. The remaining rows are then
-sequentially populated using the reducer's @pyret{.reduce} method, which is
-identical to the @pyret{.one} method except that it receives an additional
-argument which is the previously mentioned accumulated information from the
-previous row.
+A "reducing" column is one whose information is computed from the
+row it is being added to @italic{and one or more of the rows above}
+that row.  This is
+analogous to the @pyret{fold} function for @seclink{lists}.
 
-To illustrate, a @pyret{running-mean} reducer which is equivalent to the
-one provided by the @tt{tables} module could be implemented as follows:
-@pyret-block{
-import tables as TS
-running-mean :: TS.Reducer<{Number; Number}, Number, Number> = {
-  one: lam(n): {{n; 1}; n} end,
-  reduce: lam({sum; count}, n):
-    { {sum + n; count + 1}; (sum + n) / (count + 1) }
-  end
-}
+The simplest examples of reducing use reducers built into Pyret.
+
+In each reducer below, you will need to specify
+a name for the new column and which existing column new value
+will be based on.  
+
+@margin-note{To use @pyret{extend} expressions you will need to
+@pyret{import} or @pyret{include} @pyret{tables}.}
+
+@value["running-sum" (Red-of N N N)]
+
+Creates a new column where in each row,
+the running sum will be the added
+value of the cell in the selected column plus all the cells
+@italic{above} the cell in the same column.
+
+@examples{
+import tables as T
+dem-primary-delegates = table: state :: String, clinton :: Number, 
+  sanders :: Number
+  row: "Iowa", 29, 21
+  row: "New Hampshire", 15, 16
+  row: "Nevada", 27, 16
+  row: "South Carolina", 44, 14
+end
+running-total-delegates = extend dem-primary-delegates 
+  using clinton, sanders:
+  total-clinton: T.running-sum of clinton,
+  total-sanders: T.running-sum of sanders
+end
+print(running-total-delegates)
 }
 
-@subsection{Reducers}
+@(image "src/builtin/primaries.png")
 
-The following reducers are provided:
+@value["difference" (Red-of N N N)]{
 
-@value["running-mean" (Red-of N N N)]
+The @pyret{difference} extender creates a new column 
+containing the difference between the value in
+the current row (of the selected column) minus the value in @italic{only}
+the row directly above.  In the first row, zero is subtracted from the value.
```

Comment:
"In the first row, the value is unchanged (since there's no value before the first row, Pyret behaves as if it were zero)"